src/ray/core_worker/profiling.h

```diff
@@ -18,14 +18,11 @@ class Profiler {
            const std::shared_ptr<gcs::RedisGcsClient> &gcs_client);
 
   // Add an event to the queue to be flushed periodically.
-  void AddEvent(const rpc::ProfileTableData::ProfileEvent &event) LOCKS_EXCLUDED(mutex_);
+  void AddEvent(const rpc::ProfileTableData::ProfileEvent &event);
 
  private:
   // Flush all of the events that have been added since last flush to the GCS.
-  void FlushEvents() LOCKS_EXCLUDED(mutex_);
-
-  // Mutex guarding rpc_profile_data_.
-  absl::Mutex mutex_;
```

Comment:
Would you mind explain on why this mutext is no longer needed?

Reply:
Actually AddEvent is still need this lock. I will fix this.